File: 2019/02/02.py
with open("input") as file:
    data = [int(num) for num in file.read().split(',')]

def run_program(memory, noun, verb):
    memory[1], memory[2] = noun, verb
    for i in range(0, len(memory), 4):
        opcode, num1, num2, location = memory[i: i + 4]
        if opcode == 99:
            break
        elif opcode == 1:
            memory[location] = memory[num1] + memory[num2]
        elif opcode == 2:
            memory[location] = memory[num1] * memory[num2]
    return memory[0]

def find_answer():
    for i in range(100):
        for j in range(100):
            if (run_program(data[::], i, j) == 19690720):
                return 100 * i + j

print(find_answer())

File: 2019/02/02.swift
//
//  02.swift
//
//
//  Created by Brittany Pinder on 2021-08-17.
//

import Foundation

func runProgram(program: [Int], noun: Int, verb: Int) -> Int {
    var codes = program

    codes[1] = noun
    codes[2] = verb

    for (index, code) in codes.enumerated() where index % 4 == 0 {
        if code == 99 {
            break
        }

        let position1 = codes[index + 1]
        let position2 = codes[index + 2]
        let position3 = codes[index + 3]

        switch code {
            case 1:
                codes[position3] = codes[position1] + codes[position2]
            case 2:
                codes[position3] = codes[position1] * codes[position2]
            default:
                print("Unknown opcode encountered")
        }
    }

    return codes[0]
}

let contents = try! String(contentsOfFile: "Input")

let trimmedContents = contents.replacingOccurrences(of: "\n", with: "")

var codes = trimmedContents.split(separator:",")
                           .compactMap { Int($0) }

var answer: Int? = nil

for noun in 0...99 {
    for verb in 0...99 {
        if (runProgram(program: codes, noun: noun, verb: verb)) == 19690720 {
            answer = 100 * noun + verb
        }
    }
}

print(answer ?? "Answer not found!")
assert(answer == 9074, "Answer is incorrect!")
